tests/topk_varlen/test_topk_varlen_gvr2.py

```
@@ -0,0 +1,743 @@
+
+# ---------------------------------------------------------------------------
+# backend registration / guards / cross-backend
+# ---------------------------------------------------------------------------
+
+
+@requires_gvr2
+def test_gvr2_rejects_non_fp32():
+    logits = torch.randn(4, 8192, dtype=torch.bfloat16, device=_DEV)
+    seq_lens = torch.full((4,), 8192, dtype=torch.int32, device=_DEV)
+    pre_idx = torch.zeros(4, 512, dtype=torch.int32, device=_DEV)
+    with pytest.raises(Exception, match="not supported|Problem size"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)